src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const oldVersion = {};
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
```

Comment:
Consider using object destructuring for 'fromFormat' to improve code readability. For example, `const [yearKey, monthKey, dayKey, fromSeparator] = fromFormat;` can be used to directly access the format parts and separator.